test/integration/dataunion/withdraw.test.ts

```diff
@@ -131,13 +139,39 @@ const testWithdraw = async (
     const stats = await memberClient.getDataUnion(dataUnion.getAddress()).getMemberStats(memberWallet.address)
     log(`Stats: ${JSON.stringify(stats)}`)
 
-    // test setup done, do the withdraw
-    const balanceBefore = await getBalance(memberWallet)
+    const getRecipientBalance = async () => (
+        options.sendToMainnet
+            ? memberClient.getTokenBalance(recipient)
+            : memberClient.getSidechainTokenBalance(recipient)
+    )
+
+    const balanceBefore = await getRecipientBalance()
     log(`Balance before: ${balanceBefore}. Withdrawing tokens...`)
 
-    const withdrawTr = await withdraw(dataUnion.getAddress(), memberClient, memberWallet, adminClient)
-    log(`Tokens withdrawn, sidechain tx receipt: ${JSON.stringify(withdrawTr)}`)
-    const balanceAfter = await getBalance(memberWallet)
+    // "bridge-sponsored mainnet withdraw" case
+    if (!options.payForTransport && options.waitUntilTransportIsComplete) {
+        log(`Adding ${recipient} to bridge-sponsored withdraw whitelist`)
+        bridgeWhitelist.push(recipient)
+    }
+
+    // test setup done, do the withdraw
+    let ret = await withdraw(dataUnion.getAddress(), memberClient, memberWallet, adminClient)
+
+    // "other-sponsored mainnet withdraw" case
+    if (typeof ret === 'string') {
+        log(`Transporting message "${ret}"`)
+        ret = await dataUnion.transportMessage(String(ret))
+    }
+    log(`Tokens withdrawn, return value: ${JSON.stringify(ret)}`)
```

Comment:
Remember you can use string formatting tokens instead of stringify e.g. 
```js
log(`Tokens withdrawn, return value: %o,  ret)
```

Reply:
interesting, I'll try it!

Reply:
I'll just do a complete logging cleanup now